src/vm/instruction_set.md

```diff
@@ -1790,6 +1792,53 @@ If the signature cannot be verified, `MEM[$rA, 64]` is set to `0` and `$err` is
 
 To get the address from the public key, hash the public key with [SHA-2-256](#s256-sha-2-256).
 
+### ECR1: Secp256r1 signature recovery
+
+|             |                                                                                                                             |
+|-------------|-----------------------------------------------------------------------------------------------------------------------------|
+| Description | The 64-byte public key (x, y) recovered from 64-byte signature starting at `$rB` on 32-byte message hash starting at `$rC`. |
+| Operation   | ```MEM[$rA, 64] = recover(MEM[$rB, 64], MEM[$rC, 32]);```   |
+| Syntax      | `ecr $rA, $rB, $rC`         |
+| Encoding    | `0x00 rA rB rC -`   |
+| Notes       |                    |
+
+Panic if:
+
+- `$rA + 64` overflows
+- `$rB + 64` overflows
+- `$rC + 32` overflows
+- `$rA + 64 > VM_MAX_RAM`
+- `$rB + 64 > VM_MAX_RAM`
+- `$rC + 32 > VM_MAX_RAM`
+- The memory range `MEM[$rA, 64]` does not pass [ownership check](./index.md#ownership)
+
+Signatures and signature verification are specified [here](../protocol/cryptographic_primitives.md#public-key-cryptography).
+
+If the signature cannot be verified, `MEM[$rA, 64]` is set to `0` and `$err` is set to `1`, otherwise `$err` is cleared.
+
+### ED19: Ed25519 signature recovery
```

Comment:
Doing public key recovery with  `ed25519` is not possible, see [this](https://cryptobook.nakov.com/digital-signatures/eddsa-and-ed25519) for instance.

Reply:
Should be `verification` not `signature recovery`. My bad.

Reply:
It's not recovering the public key, it should just be verifying the signature against the public key and message.